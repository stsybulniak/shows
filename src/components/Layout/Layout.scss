.Layout {
  background-color: white;
  height: 100vh;
  width: 100vw;
  &__main {
    height: calc(100% - 10rem);
    overflow-y: auto;

    &-content {
      min-height: calc(100% - 10rem);
      padding: 4rem;
    }
  }
}