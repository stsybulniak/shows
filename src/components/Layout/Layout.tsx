import { FC } from 'react';
import Footer from './Footer/Footer';
import Header from './Header/Header';
import './Layout.scss';

const Layout: FC = ({ children }) => (
  <div className='Layout'>
    <Header />
    <main className='Layout__main'>
      <div className='Layout__main-content'>{children}</div>
      <Footer />
    </main>
  </div>
);

export default Layout;
