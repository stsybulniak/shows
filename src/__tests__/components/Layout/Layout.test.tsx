import { render } from '@testing-library/react';
import Layout from '../../../components/Layout/Layout';

describe('<Layout />', () => {
  it('Layout matches snapshot', () => {
    const { baseElement } = render(<Layout><b>Layout content</b></Layout>);
    expect(baseElement).toMatchSnapshot();
  });
})
