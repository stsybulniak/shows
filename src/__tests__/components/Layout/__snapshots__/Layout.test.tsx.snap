// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> Layout matches snapshot 1`] = `
<body>
  <div>
    <div
      class="Layout"
    >
      <header
        class="Header"
      >
        Header
      </header>
      <main
        class="Layout__main"
      >
        <div
          class="Layout__main-content"
        >
          <b>
            Layout content
          </b>
        </div>
        <footer
          class="Footer"
        >
          Footer
        </footer>
      </main>
    </div>
  </div>
</body>
`;
